### PR TITLE
kubeadm: Add (duplicated) v1alpha2 Config API

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
Work in progress PR to add a (initially duplicated) `v1alpha2` we can iterate on during the cycle.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes/community/pull/2131

**Special notes for your reviewer**:
This PR depends on:
 - [x] #63782
 - [x] #63783
 - [x] #63787
 - [x] #63799

The first commit is from #63799.
The second commit duplicates v1alpha1, but updates timestamps, and doesn't require the `upgrade.go`.
The third commit does the mechanical bump of using v1alpha1 -> v1alpha2
The fourth commit updates bazel

**Release note**:

```release-note
[action required] kubeadm now uses an upgraded API version for the configuration file, `kubeadm.k8s.io/v1alpha2`. kubeadm in v1.11 will still be able to read `v1alpha1` configuration, and will automatically convert the configuration to `v1alpha2` internally and when storing the configuration in the ConfigMap in the cluster.
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @liztio

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/v1alpha2/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
+/*
```
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha2/upgrade_test.go]: probably don't need to copy this over? or if we do, we should delete it from v1alpha1

File: cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	// Setting .CloudProvider is not supported from internal API not supported
+
+	// TODO: Add more
+
+	return nil
+}
+
+func Convert_v1alpha1_MasterConfiguration_To_kubeadm_MasterConfiguration(in *MasterConfiguration, out *kubeadm.MasterConfiguration, s conversion.Scope) error {
```
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion.go]: does this handle the proxy featureflags?

**Comment by liztio** [comment]
this may be easier to read once #63782 is rebased onto it

File: cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion.go (hunk 2 above)
Comment (luxas) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion.go]: No, `Migrate` takes care of that, as that happens at unmarshal time, not conversion time

File: cmd/kubeadm/app/apis/kubeadm/v1alpha2/upgrade_test.go (hunk 1 above)
Comment (luxas) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha2/upgrade_test.go]: oh, I will delete this. This isn't needed in v1alpha2

**Comment by luxas** [comment]
> this may be easier to read once #63782 is rebased onto it

yes, very much easier after the first PRs are merged.

**Review by liztio — APPROVED** [review]
I think next time it would've been better to add the code in one PR, then switch it as the default in the next one. But I understand we were under a time crunch.

**Comment by luxas** [comment]
This is ready to go, if you only review https://github.com/kubernetes/kubernetes/pull/63788/commits/9a264307914a92675f3574f208054bb33306f2ae
The first commit is #63799 and the others autogenerated stuff

**Comment by luxas** [comment]
Readding the LGTM label, I just rebased upon #63799.
Readding approved label as well as I had to add the API package to `hack/.golint_failures`. Unnecessary to ask for approval from `hack/OWNERS` for that.

**Pull request merged**